app/client/src/components/shared/MapWidgets.tsx

```diff
@@ -1094,38 +1109,50 @@ function MapWidgets({
 }
 
 function ShowAddSaveDataWidget({
-  setAddSaveDataWidgetVisibleParam,
+  addSaveDataWidgetVisible,
+  setAddSaveDataWidgetVisible,
 }: {
-  setAddSaveDataWidgetVisibleParam: Dispatch<SetStateAction<boolean>>;
+  addSaveDataWidgetVisible: boolean;
+  setAddSaveDataWidgetVisible: Dispatch<SetStateAction<boolean>>;
 }) {
   const [hover, setHover] = useState(false);
 
-  const widget = document.getElementById('add-save-data-widget');
-  const widgetHidden = widget?.classList.contains('hidden');
+  const clickHandler = useCallback(
+    (ev: React.MouseEvent | React.KeyboardEvent) => {
+      if (!isClick(ev)) return;
+      const widget = document.getElementById('add-save-data-widget');
+      if (!widget) return;
+
+      const widgetHidden = widget.classList.contains('hidden');
+      if (widgetHidden) {
+        widget.classList.remove('hidden');
+        setAddSaveDataWidgetVisible(true);
+      } else {
+        widget.classList.add('hidden');
+        setAddSaveDataWidgetVisible(false);
+      }
+    },
+    [setAddSaveDataWidgetVisible],
+  );
 
   return (
     <div
       className="add-save-data-widget"
       title={
-        widgetHidden
+        addSaveDataWidgetVisible
```

Comment:
```suggestion
        !addSaveDataWidgetVisible
```

This is backwards. The tooltip says "Open...Widget" when the widget is open and "Close...Widget" when the widget is closed.

Reply:
Oops, good catch!